__init__.py

```diff
@@ -161,28 +161,36 @@ def initialize(self):
         self.alt_feed_names = self.translate_namedvalues('alt.feed.name')
 
     def CPS_match_query_phrase(self, phrase):
+        matched_feed = { 'name': None, 'conf': 0.0 }
         # Look for a specific news provider
-        def match_feed_name(phrase, feed_name):
-            return (fuzzy_match(phrase, feed_name) > 0.7 or
-                    fuzzy_match(phrase, feed_name + self.translate("News")) > 0.7)
-
-        phrase = ' '.join(phrase.lower().split())
-        news_voc = self.translate("News") if self.voc_match(phrase, "News") else ""
+        def match_feed_name(phrase, source, alt_feed=None):
+            nonlocal matched_feed
+            phrase, feed_name = phrase.lower(), source.lower()
+            # Test with News added in case user only says acronym eg "ABC".
+            # As it is short it may not provide a high match confidence.
+            conf = max(fuzzy_match(phrase, feed_name),
+                    fuzzy_match(phrase, feed_name + self.translate("News")))
+            if conf > matched_feed['conf']:
+                matched_feed = { 'name': alt_feed or source, 'conf': conf }
 
         # Check primary feed list for matches eg 'ABC'
         for source in FEEDS:
-            if match_feed_name(phrase, source.lower()):
-                return (source + news_voc, CPSMatchLevel.EXACT,
-                        {"feed": source})
+            match_feed_name(phrase, source)
+
         # Check list of alternate names eg 'associated press' => 'AP'
         for name in self.alt_feed_names:
-            if match_feed_name(phrase, name.lower()):
-                return (self.alt_feed_names[name] + " " + news_voc,
-                        CPSMatchLevel.EXACT,
-                        {"feed": self.alt_feed_names[name]})
-
-        if self.voc_match(phrase, "News"):
-            return ("news", CPSMatchLevel.TITLE)
+            match_feed_name(phrase, name, self.alt_feed_names[name])
```

Comment:
Should the function return the name and confidence instead of passing through a shared variable?

Making it something like
```python
        def match_feed_name(phrase, source, alt_feed=None):                     
            phrase, feed_name = phrase.lower(), source.lower()                  
            # Test with News added in case user only says acronym eg "ABC".     
            # As it is short it may not provide a high match confidence.        
            conf = max(fuzzy_match(phrase, feed_name),                          
                    fuzzy_match(phrase, feed_name + self.translate("News")))
           return alt_feed or source, conf
                                                                  
        # Check primary feed list for matches eg 'ABC'                          
        for source in FEEDS:                                                    
            name, conf = match_feed_name(phrase, source)
            if conf > matched_feed['conf']:
                matched_feed['conf'] = conf
                matched_feed['name'] = name
```

Reply:
Good call, changed.